tests/data_association/test_data_association.py

```diff
@@ -252,13 +252,20 @@ def test_data_association_with_missing_camera(self):
             2: PinholeCameraCal3Bundler(Pose3(Rot3.RzRyRx(0, 0, 0), np.array([10, 0, 0]))),
         }
 
-        # just have one track
+        # just have one track, chaining cams 0->1 , and cams 1->2
         corr_idxs_dict = {(0, 1): np.array([[0, 0]], dtype=np.int32), (1, 2): np.array([[0, 0]], dtype=np.int32)}
         keypoints_shared = Keypoints(coordinates=np.array([[20.0, 10.0]]))
 
-        sfm_data, _ = da.run(3, cameras, corr_idxs_dict, [keypoints_shared] * 3)
+        # will lead to a cheirality exception because keypoints are identical in two cameras
+        # no track will be formed, and thus connected component will be empty
+        sfm_data, _ = da.run(
+            num_images=3,
+            cameras=cameras,
+            corr_idxs_dict=corr_idxs_dict,
+            keypoints_list=[keypoints_shared] * 3
+        )
 
-        self.assertEqual(len(sfm_data.get_valid_camera_indices()), 2)
+        self.assertEqual(len(sfm_data.get_valid_camera_indices()), 0)
```

Comment:
this test should check if 0->1->2, right? Maybe we need to have a new test for the cheirality exception.

Reply:
Hmm I didn't quite catch what you meant here @ayushbaid

Reply:
the cheirality exception is being caught here still

Reply:
I mean why did the excepted value change from 2 to 0.

Reply:
thanks for the review @ayushbaid. Basically we have just one track, but since there is a cheirality exception when we try to process that one track, there is no edge between any of the cameras. This means that 0 cameras end up in the largest CC. This is the reason for the change.

Reply:
```
keypoints_list = [
            feature_utils.generate_random_keypoints(1, [200, 200]),
            feature_utils.generate_random_keypoints(1, [200, 200]),
            feature_utils.generate_random_keypoints(1, [200, 200]),
        ]
```